src/mds/MDCache.cc

```diff
@@ -12311,10 +12311,10 @@ void MDCache::rollback_uncommitted_fragment(dirfrag_t basedirfrag, frag_vec_t&&
   }
 }
 
-void MDCache::wait_for_uncommitted_fragments(MDSGather *gather)
+void MDCache::wait_for_uncommitted_fragments(MDSGatherBuilder& gather)
 {
```

Comment:
it's better to define MDSGatherBuilder here. and make this function accept a MDSInternalContext